source/fuzz/transformation_wrap_vector_synonym.cpp

```diff
@@ -0,0 +1,151 @@
+  message_.set_fresh_id(fresh_id);
+  message_.set_scalar_position(pos);
+}
+
+bool TransformationWrapVectorSynonym::IsApplicable(
+    opt::IRContext* ir_context,
+    const TransformationContext& transformation_context) const {
+  auto instruction =
+      ir_context->get_def_use_mgr()->GetDef(message_.instruction_id());
+  auto vec1 = ir_context->get_def_use_mgr()->GetDef(message_.vector_operand1());
+  auto vec2 = ir_context->get_def_use_mgr()->GetDef(message_.vector_operand2());
+
+  // |instruction_id| must refer to an existing instruction.
+  if (instruction == nullptr) {
+    return false;
+  }
+
+  auto type_instruction =
+      ir_context->get_def_use_mgr()->GetDef(instruction->type_id());
+
+  // The instruction must be of a valid scalar operation type.
+  if (!OpcodeIsSupported(instruction->opcode()) ||
+      !OperandTypeIsSupported(type_instruction)) {
+    return false;
+  }
+
+  assert(!transformation_context.GetFactManager()->IdIsIrrelevant(
+             instruction->result_id()) &&
+         "Result id of the scalar operation must be relevant.");
+
+  // |vector_operand1| and |vector_operand2| must exist.
+  if (vec1 == nullptr || vec2 == nullptr) {
+    return false;
+  }
+
+  // The 2 vectors must be the same valid vector type.
+  auto vec1_type_id = vec1->type_id();
+  auto vec2_type_id = vec2->type_id();
+
+  if (vec1_type_id != vec2_type_id) {
+    return false;
+  }
+
+  auto vec_type = ir_context->get_def_use_mgr()->GetDef(vec1_type_id)->opcode();
+
+  if (vec_type != SpvOpTypeVector) {
+    return false;
+  }
+
+  // |fresh_id| must be fresh.
+  if (!fuzzerutil::IsFreshId(ir_context, message_.fresh_id())) {
+    return false;
+  }
+
+  // |scalar_position| needs to be a non-negative integer less than the vector
+  // length.
+  // OpTypeVector instruction has the component count at index 2.
+  auto vec_len = ir_context->get_def_use_mgr()
+                     ->GetDef(vec1_type_id)
+                     ->GetSingleWordInOperand(0);
+  if (message_.scalar_position() >= vec_len) {
+    return false;
+  }
+
+  return true;
+}
+
+void TransformationWrapVectorSynonym::Apply(
+    opt::IRContext* ir_context,
+    TransformationContext* transformation_context) const {
+  // Create an instruction descriptor for the original instruction.
+  auto instruction =
+      ir_context->get_def_use_mgr()->GetDef(message_.instruction_id());
+  auto destination_block = ir_context->get_instr_block(instruction);
+
+  //  Populate input operand list with two vectors for vector operation.
+  opt::Instruction::OperandList in_operands;
+  in_operands.push_back({SPV_OPERAND_TYPE_ID, {message_.vector_operand1()}});
+  in_operands.push_back({SPV_OPERAND_TYPE_ID, {message_.vector_operand2()}});
+
+  // Make a new arithmetic instruction: %fresh_id = OpXX %type_id %result_id1
+  // %result_id2.
+  auto vec_type_id = ir_context->get_def_use_mgr()
+                         ->GetDef(message_.vector_operand1())
+                         ->type_id();
```

Comment:
This is an example of a temporary that is only used once, but that I *do* think makes the code more readable (i.e., keep it.)

Reply:
Got it. Thanks!